metadata-ingestion/src/datahub/sdk/entity_client.py

```diff
@@ -164,9 +165,16 @@ def upsert(self, entity: Entity) -> None:
             # TODO: If there are no previous aspects but the entity exists, should we delete aspects that are not present here?
 
         mcps = entity.as_mcps(models.ChangeTypeClass.UPSERT)
-        self._graph.emit_mcps(mcps)
+        if emit_mode is not None:
+            self._graph.emit_mcps(mcps, emit_mode=emit_mode)
+        else:
+            self._graph.emit_mcps(mcps)
 
-    def update(self, entity: Union[Entity, MetadataPatchProposal]) -> None:
+    def update(
+        self,
+        entity: Union[Entity, MetadataPatchProposal],
+        emit_mode: Optional[EmitMode] = None,
```

Comment:
make this a keyword-only arg
```suggestion
        *, emit_mode: Optional[EmitMode] = None,
```